finagle-thrift/src/main/ruby/lib/finagle-thrift/trace.rb

```diff
@@ -57,6 +57,10 @@ def tracer=(tracer)
     @tracer = tracer
   end
 
+  def stack=(stack)
```

Comment:
I don't know ruby, so please be patient with me :)

It looks like this is sticking the stack into a thread-local variable.  Is our guarantee for ruby tracing that you won't ever switch threads?